pom.xml

```diff
@@ -1324,5 +1324,21 @@
                 <lucene-dependency-scope>provided</lucene-dependency-scope>
             </properties>
         </profile>
+        <profile>
+            <id>fail-silent-on-no-git-dir</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>pl.project13.maven</groupId>
+                        <artifactId>git-commit-id-plugin</artifactId>
+                        <version>4.0.0</version>
+                        <configuration>
```

Comment:
I am not sure if the profile plugin configuration will overwrite the default configuration or merge into the default configuration.

As you just added the new configurations, it may seem to be they will be merged into the default configuration, do we have any documentation about profile behavior about this?

Reply:
the configuration in profile will overwrite these in the build configuration.  the priority for the configuration is: `configuration in profile(after enabled)` > `build configuration`.

can ref the resource[1][2] here.
[1] https://maven.apache.org/guides/introduction/introduction-to-profiles.html?spm=5176.28103460.0.0.6077572cRnkc6p#profile-inheritance
[2] https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Project_Inheritance